powersimdata/scenario/execute.py

```diff
@@ -203,9 +207,30 @@ def __init__(self, data_access, scenario_info, grid, ct):
 
         self.REL_TMP_DIR = self._data_access.tmp_folder(self.scenario_id)
 
+    def _adjust_pmin(self):
+        mi = self.grid.model_immutables
+        plant = self.grid.plant
+        pmin_factor = mi.plants["pmin_as_share_of_pmax"]
+
+        def _scale(x):
+            factor = pmin_factor.get(x.type)
+            return x.Pmin if factor is None else factor * x.Pmax
+
+        plant.Pmin = plant.apply(_scale, axis=1)
+
+        # set Pmin to 0 for generators that are off or profile based
+        profile_resource = list(mi.plants["profile_resources"])
+        plant.loc[plant.type.isin(profile_resource), "Pmin"] = 0
+        plant.loc[plant.status == 0, "Pmin"] = 0
+
     def prepare_grid(self):
         """Prepare grid for simulation."""
         print("--> Preparing grid data")
+        self._adjust_pmin()
+        storage = self.grid.storage
+        self.grid.gencost["after"] = linearize_gencost(self.grid)
+        storage["gencost"] = _linearize_gencost(storage["gencost"], storage["gen"])
```

Comment:
I am not sure I understand why we need two functions. Could we do:
```
storage["gencost"] = _linearize_gencost(storage["gencost"], storage["gen"])
self.grid.gencost["after"] = _linearize_gencost(elf.grid.gencost["before"], self.grid.plant)
```
Then we could discard the `linearize_gencost` function in the `cost_curves` module and rename `_linearize_gencost` to `linearize_gencost`

Reply:
done